java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -470,6 +474,27 @@ public static HostColumnVector fromStrings(String... values) {
     });
   }
 
+  /**
+   * Create a new vector from the given values.  This API supports inline nulls, but it is inefficient.
+   * Notice all input BigDecimals should share same scale.
+   */
+  public static HostColumnVector fromDecimals(BigDecimal... values) {
+    // Try to fetch the element with max precision. Fill with ZERO if inputs is empty.
+    BigDecimal maxPrecisionDec = Arrays.stream(values).filter(Objects::nonNull)
+        .max(Comparator.comparingInt(BigDecimal::precision))
+        .orElse(BigDecimal.ZERO);
+    // Select appropriate underlying type according to max precision.
```

Comment:
This still loops through all the values to find the maximum scale value when we only care about finding any non-null scale value. Something like this would be more appropriate:
```suggestion
    BigDecimal decimal = Arrays.stream(values).filter(Objects::nonNull)
        .findAny()
        .orElse(BigDecimal.ZERO);
    // Select appropriate decimal type according to the decimal's precision.
```

Reply:
I think we still need to know the maximum precision, which is helpful to determine the underlying storage type (DECIMAL32 or DECIMAL64).